### PR TITLE
feat(mappings): considering map description when filtering

### PR DESCRIPTION
**Pull request by shadmansaleh**
Allows filtering map output by map description.

### CODE REVIEW

**Comment by zeertzjq** [comment]
Actually filtering `<Nop>` can be done using `:filter /^$/ map`.

**Comment by shadmansaleh** [comment]
> Actually filtering <Nop> can be done using :filter /^$/ map.

You're right. Then <Nop> can be dropped . What about <Cmd> / <Lua> ?

**Comment by zeertzjq** [comment]
`:filter /\// map`?

**Comment by shadmansaleh** [comment]
> :filter /\// map?

Sure didn't think about escaping in pattern. Then just adding the desc based filter should be enough.

File: src/nvim/getchar.c [hunk 1]
```diff
@@ -3435,7 +3435,8 @@ static void showmap(mapblock_T *mp, bool local)
   size_t len = 1;
 
   if (message_filtered(mp->m_keys)
-      && mp->m_str != NULL && message_filtered(mp->m_str)) {
+      && (mp->m_str[0] == NUL || message_filtered(mp->m_str))
```
Comment (zeertzjq) [inline src/nvim/getchar.c:3438]: Why add this `mp->m_str[0] == NUL` check?

Comment (shadmansaleh) [reply src/nvim/getchar.c:3437]: Not required I'll remove it

Comment (shadmansaleh) [reply src/nvim/getchar.c]: Removed.

**Comment by zeertzjq** [comment]
Did you test with a Lua mapping?

**Comment by zeertzjq** [comment]
Also I think this should be `feat: `.

**Comment by zeertzjq** [comment]
`mp->m_str` can never be `NULL`. I think you can remove the check in `map_to_exists_mode()` as well.

**Comment by zeertzjq** [comment]
I'll do that in another PR.

**Pull request merged**